internal/ingest/linksystem.go

```diff
@@ -191,29 +192,72 @@ func (ing *Ingester) ingestAd(publisherID peer.ID, adCid cid.Cid, ad schema.Adve
 		}
 	}
 
-	var maddrs []multiaddr.Multiaddr
-	if len(ad.Addresses) != 0 {
-		maddrs = make([]multiaddr.Multiaddr, 0, len(ad.Addresses))
-		for _, addr := range ad.Addresses {
-			maddr, err := multiaddr.NewMultiaddr(addr)
-			if err != nil {
-				log.Warnw("Bad address in advertisement", "address", addr)
-				continue
-			}
-			maddrs = append(maddrs, maddr)
-		}
-	}
+	maddrs := stringsToMultiaddrs(ad.Addresses)
 
 	provider := peer.AddrInfo{
 		ID:    providerID,
 		Addrs: maddrs,
 	}
-	err = ing.reg.RegisterOrUpdate(context.Background(), provider, adCid, publisher)
+
+	ctx := context.Background()
+
+	err = ing.reg.RegisterOrUpdate(ctx, provider, adCid, publisher)
 	if err != nil {
 		return adIngestError{adIngestRegisterProviderErr, fmt.Errorf("could not register/update provider info: %w", err)}
 	}
 
-	log = log.With("provider", providerID)
+	if ad.ExtendedProvider != nil {
+
+		if ad.IsRm {
+			return adIngestError{adIngestIndexerErr, fmt.Errorf("rm ads can not have extended providers")}
+		}
+
+		if len(ad.ContextID) == 0 && ad.ExtendedProvider.Override {
+			return adIngestError{adIngestIndexerErr, fmt.Errorf("override can not be set on extended provider without context id")}
```

Comment:
Since the ad is not being ingested, it probably makes sense to call `RegisterOrUpdate` _after_ knowing the ad is acceptable. As per the comment above, it also makes sense to register all the information from the ad in one call to the registry.

Reply:
Good catch, fixed